analysis.go

```diff
@@ -79,3 +80,49 @@ func (g AnalyzerCommentsGroups) Filter(fn CommentsFilterFn) ([]AnalyzerComments,
 
 	return result, nil
 }
+
+// Dedup filters duplicated comments
+func (g AnalyzerCommentsGroups) Dedup() AnalyzerCommentsGroups {
+	var result []AnalyzerComments
+	type Key struct {
+		File         string
+		Line         int32
+		Text         string
+		AnalyzerName string
+	}
+
+	uniqueCommentsMap := map[Key]*Comment{}
```

Comment:
Can we assume @carlosms that each group contains the comments for each different analyzer?
If so, the index map `uniqueCommentsMap` could be moved into the first loop, and the `Key.AnalyzerName` property would be unnecessary.

Reply:
you're right @dpordomingo, there's no more need to initialize it outside the outer loop 👍

Reply:
Addressed [here](https://github.com/src-d/lookout/pull/458/commits/78381b43a61fe204a144b424244a625fa7c25ac3)